lib/ueberauth/strategy.ex

```diff
@@ -318,4 +339,80 @@ defmodule Ueberauth.Strategy do
     auth = apply(strategy, :auth, [conn])
     Plug.Conn.assign(conn, :ueberauth_auth, auth)
   end
+
+  defp state_param_matches?(conn) do
+    state_cookie = get_state_cookie(conn)
+    param_cookie = conn.params["state"]
+
+    case {state_cookie, param_cookie} do
+      {nil, nil} -> false
+      {value, value} -> true
+      {_, _} -> false
+    end
+  end
```

Comment:
Just a suggestion you can pass on but we could possibly clean this up a little:

```suggestion
  defp state_param_matches?(conn) do
    param_cookie = conn.params["state"]
    not is_nil(param_cookie) and param_cookie == get_state_cookie(conn)
  end
```